teps/0046-pipelinerun-in-a-pod.md

```diff
@@ -0,0 +1,97 @@
+that can be shared between them.
+
+Some problems with this:
+* PVCs add additional overhead, both in speed, in resource costs and in management.
+* PVCs is hard to share between multiple Pods concurrently - needs appropriate scheduling of the Pods.
+* PVCs is difficult to share in subsequently run Pods that might be scheduled to a different AZ than where the
+  volume is located.
+
+### Goals
+
+- Make it possible to run whole PipelineRun (including all TaskRuns) within a single Pod so that the workspace can be within the Pod:
+    - At runtime in a PipelineRun
+    - This should not change anything for the author of Tasks or Pipelines
+- No changes in the Pipeline or Task API for authors
+
+### Non-Goals
+
+- To fully handle the Pod resource requests and limits in an appropriate way for the PipelineRun Pod.
+
+### Use Cases (optional)
+
+- A user want to create a typical Build pipeline with git-clone, lint, code-build, unit-tests and multiple
+  parallel integration tests - all Tasks sharing the volume that was populated by git-clone and read by all
+  subsequent tasks to perform "validation". The user would like to see its PipelineRun run to completion as
+  a single unit INSTEAD OF worry about appropriately scheduled taskRun Pods in a busy cluster and need to
+  use Persistent Volumes for the files from git-clone.
+- A user want to run a Build pipeline that starts with two concurrent tasks, git-clone and another task that
+  fetches cached dependencies from a cache and then build the cloned code while reusing the cached dependencies
+  to speed up the build and see the full PipelineRun to be run in a single Pod, on a single Node WITHOUT NEEDING
+  to mount external volumes and worry that its Pipeline is composed of several Pods that might be scheduled to 
+  different cloud Availability Zones and reach a deadlocked state.
+- A user wants to create a Pipeline composed of Tasks from the catalog, to checkout code, run unit tests and upload results,
+  and does not want to incur the additional overhead (and performance impact) of creating
+  volume based workspaces to share data between them in a Pipeline. e.g. specifically
+  cloning with [git-clone](https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.2),
+  running tests with [golang-test](https://github.com/tektoncd/catalog/tree/master/task/golang-test/0.1)
+  and uploading results with [gcs-upload](https://github.com/tektoncd/catalog/tree/master/task/gcs-upload/0.1).
+
+## Requirements
+
+- Pipeline can be composed of multiple Tasks and scheduled to run together in a single Pod:
+    - Must be able to share data without requiring
+      an external volume (i.e. probably the containers that make them up are run within
+      the same pod)
+- Tasks should be able to run concurrently if the Pipeline author want that
+- All Tasks should run even if one fails
+    - This is to support use cases such as uploading test results, even if the test
+      Task failed
```

Comment:
I am not sure I follow this requirements. Why would we treat tasks in a pipeline that runs into a single `Pod` differently than when it runs into a as of today ? For example, supporting the use case of uploading test results, can be handled using `finally` task today, it would be the same there. 

Except running in the same pod and thus removing some overhead, I think we should limit the difference in assumption and "how things run" between running into a single pod or several.